src/collection.ts

```diff
@@ -710,16 +709,10 @@ export class Collection<TSchema extends Document = Document> {
     if (typeof options === 'function') (callback = options), (options = {});
 
     filter ??= {};
-
-    return executeOperation(
-      getTopology(this),
-      new FindOneOperation(
-        this as TODO_NODE_3286,
-        filter,
-        resolveOptions(this, options)
-      ) as TODO_NODE_3286,
-      callback as TODO_NODE_3286
-    );
+    return this.find(filter, options)
+      .limit(-1)
+      .batchSize(1)
+      .next(callback as TODO_NODE_3286);
```

Comment:
(Just an fyi cus I was perusing the CI) Looks like the only issue is this used to return undefined, so we just need a small arrow function here to preserve that.